src/layers/image-label-layer.ts

```diff
@@ -0,0 +1,30 @@
+import { MultiscaleImageLayer } from "@hms-dbmi/viv";
+import type { Matrix4 } from "math.gl";
+
+import type { ZarrPixelSource } from "../ZarrPixelSource";
+
+export type ImageLabelLayerProps = {
+  id: string;
+  loader: Array<ZarrPixelSource>;
+  selection: Array<number>;
+  opacity: number;
+  modelMatrix?: Matrix4;
+  color?: [r: number, g: number, b: number];
+};
+
+// just subclass viv for now
+export default class ImageLabelLayer extends MultiscaleImageLayer<string[]> {
+  constructor(props: ImageLabelLayerProps) {
+    super({
+      id: `labels-${props.id}`,
+      loader: props.loader,
+      opacity: props.opacity,
+      // @ts-expect-error - Viv types aren't great
+      selections: [props.selection],
+      colors: [props.color ?? [255, 255, 255]],
+      channelsVisible: [true],
+      contrastLimits: [[0, 1]],
+      modelMatrix: props.modelMatrix,
+    });
+  }
+}
```

Comment:
This is just a placeholder for now. The idea is to define a custom Deck.gl layer here, either extending something from Viv or creating a more tailored solution. The current plumbing allows `Labels.tsx` to control `ImageLabelLayerProps` for whatever we implement.